internal/service/ec2/vpc_endpoint_private_dns_verification.go

```diff
@@ -0,0 +1,105 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkResource("aws_vpc_endpoint_service_private_dns_verification", name="Endpoint Service Private DNS Verification")
+func newResourceEndpointServicePrivateDNSVerification(_ context.Context) (resource.ResourceWithConfigure, error) {
+	return &resourceEndpointServicePrivateDNSVerification{}, nil
+}
+
+const (
+	ResNameEndpointServicePrivateDNSVerification = "Endpoint Service Private DNS Verification"
+)
+
+type resourceEndpointServicePrivateDNSVerification struct {
+	framework.ResourceWithConfigure
+}
+
+func (r *resourceEndpointServicePrivateDNSVerification) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_vpc_endpoint_service_private_dns_verification"
+}
+
+func (r *resourceEndpointServicePrivateDNSVerification) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"service_id": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+func (r *resourceEndpointServicePrivateDNSVerification) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	conn := r.Meta().EC2Client(ctx)
+
+	var plan resourceEndpointServicePrivateDNSVerificationData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	in := &ec2.StartVpcEndpointServicePrivateDnsVerificationInput{
+		ServiceId: aws.String(plan.ServiceID.ValueString()),
+	}
+
+	out, err := conn.StartVpcEndpointServicePrivateDnsVerification(ctx, in)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.EC2, create.ErrActionCreating, ResNameEndpointServicePrivateDNSVerification, plan.ServiceID.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	if out == nil || out.ReturnValue == nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.EC2, create.ErrActionCreating, ResNameEndpointServicePrivateDNSVerification, plan.ServiceID.String(), nil),
+			errors.New("empty output").Error(),
+		)
+		return
+	}
+	if !aws.ToBool(out.ReturnValue) {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.EC2, create.ErrActionCreating, ResNameEndpointServicePrivateDNSVerification, plan.ServiceID.String(), nil),
+			errors.New("request failed").Error(),
+		)
+		return
+	}
+
```

Comment:
Does it makes sense to poll on [`PrivateDnsNameConfiguration.State` == `"verified"`](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_PrivateDnsNameConfiguration.html)?

Reply:
Because this requires some other DNS registration bits which may or may not use AWS Route53, my initial thinking was not to poll. Maybe it makes sense to add an optional argument (`wait_for_verified_status`?) so users can opt-in to this behavior if desired?

I'll think on this one a bit more.

Reply:
Started implementing an optional `wait_for_verification` argument, and realized this can share the finder from the `aws_vpc_endpoint_service` resource. I have another WIP PR that migrates this resource to AWS SDK V2, which will make it usable for this purpose. I'm going to merge this initial implementation without that argument, then add it as a follow up once the migration work is complete.

Reply:
Added an optional `wait_for_verification` argument.